compiler/src/dmd/common/string.d

```diff
@@ -13,35 +13,43 @@ module dmd.common.string;
 nothrow:
 
 /**
-Defines a temporary array using a fixed-length buffer as back store. If the length
+Defines a temporary array of `Element`s using a fixed-length buffer as back store. If the length
 of the buffer suffices, it is readily used. Otherwise, `malloc` is used to
 allocate memory for the array and `free` is used for deallocation in the
 destructor.
 
 This type is meant to use exclusively as an automatic variable. It is not
 default constructible or copyable.
 */
-struct SmallBuffer(T)
+struct SmallBuffer(Element)
 {
     import core.stdc.stdlib : malloc, free;
 
-    private T[] _extent;
+    private Element[] _extent;
     private bool needsFree;
 
   nothrow:
+  @nogc:
 
     @disable this(); // no default ctor
-    @disable this(ref const SmallBuffer!T); // noncopyable, nonassignable
-
-    this(size_t len, T[] buffer)
+    @disable this(ref const SmallBuffer!Element); // noncopyable, nonassignable
+
+    /***********
+     * Construct a SmallBuffer
+     * Params:
+     *  len = number of elements in array
+     *  buffer = slice to use as backing-store, if len will fit in it
+     */
+    scope this(size_t len, return scope Element[] buffer)
     {
         if (len <= buffer.length)
         {
             _extent = buffer[0 .. len];
         }
         else
         {
-            _extent = (cast(typeof(_extent.ptr)) malloc(len * _extent[0].sizeof))[0 .. len];
+            assert(len < sizeof.max / Element.sizeof);
```

Comment:
`sizeof.max` is bogus, you meant `size_t`. Not sure how this passed CI, or why it hasn't bee a problem until now.